src/components/ec/ucc_ec.c

```diff
@@ -102,7 +102,7 @@ ucc_status_t ucc_ec_finalize()
     ucc_ee_type_t  et;
     ucc_ec_base_t *ec;
 
-    for (et = UCC_EE_CPU_THREAD; et < UCC_EE_LAST; et++) {
+    for (et = UCC_EE_CUDA_STREAM; et < UCC_EE_LAST; et++) {
```

Comment:
actually we need to call finalize for every EC component including UCC_EE_CPU_THREAD, but for historical reasons UCC_EE_CUDA_STREAM is first EE (with id 0) declared in ucc.h. I would like to change it, but it will break API/ABI.

Reply:
that would make it even simpler, having a UCC_EE_FIRST marker type

Reply:
I was thinking about a solution like this here, which would not break the ABI
https://github.com/openucx/ucx/blob/master/src/ucp/core/ucp_types.h#L163

Reply:
I agree with Edgar, lets define FIRST/LAST markers